synthesizer/src/process/stack/execution/mod.rs

```diff
@@ -133,4 +179,9 @@ impl<N: Network> Execution<N> {
     pub fn commitments(&self) -> impl '_ + Iterator<Item = &Field<N>> {
         self.transitions.values().flat_map(Transition::commitments)
     }
+
+    /// Inserts the global_state_root.
+    pub fn add_global_state_root(&mut self, global_state_root: N::StateRoot) {
```

Comment:
nit: "set" or "update" might be more fitting than "add" here

Reply:
Done: 79f705e58